client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -28,442 +26,323 @@ import {
-			translate(
-				'Domain registration and billing will be moved from your current provider to WordPress.com'
-			),
-			translate( 'Manage your domain and site from your WordPress.com dashboard' ),
-			translate( 'Extends registration by one year' ),
-			this.getTransferFreeText(),
-			this.getTransferSalePriceText(),
-			this.getTransferPriceText(),
-		];
-		const buttonText = translate( 'Transfer to WordPress.com' );
-		const learnMore = translate( '{{a}}Learn more about domain transfers{{/a}}', {
-			components: {
-				a: (
-					<a
-						href={ localizeUrl( INCOMING_DOMAIN_TRANSFER ) }
-						rel="noopener noreferrer"
-						target="_blank"
-					/>
-				),
-			},
-		} );
 
-		return this.renderOptionContent( {
-			image,
-			title,
-			reasons,
-			onClick: this.goToTransferDomainStep,
-			buttonText,
-			isPrimary: true,
-			learnMore,
-		} );
-	};
+		let transferDomainStepURL;
 
-	renderSelectMapping = () => {
-		const { translate } = this.props;
-		const image = themesImage;
-		const title = translate( 'Map your domain without moving it from your current registrar.' );
-		const reasons = [
-			translate( 'Domain registration and billing will remain at your current provider' ),
-			translate( 'Manage some domain settings at your current provider and some at WordPress.com' ),
-			// translate( 'Continue paying for your domain registration at your current provider' ),
-			translate( "Requires changes to the domain's DNS" ),
-			this.getMappingPriceText(),
-		];
-		const buttonText = translate( 'Map your domain' );
-		const learnMore = translate( '{{a}}Learn more about domain mapping{{/a}}', {
-			components: {
-				a: (
-					<a
-						href={ localizeUrl( MAP_EXISTING_DOMAIN ) }
-						rel="noopener noreferrer"
-						target="_blank"
-					/>
-				),
-			},
-		} );
+		if ( transferDomainUrl ) {
+			transferDomainStepURL = transferDomainUrl;
+		} else {
+			const basePathForTransfer = basePath.endsWith( '/use-your-domain' )
+				? basePath.substring( 0, basePath.length - 16 )
+				: basePath;
+
+			transferDomainStepURL = `${ basePathForTransfer }/transfer`;
+
+			if ( selectedSite ) {
+				const query = stringify( {
+					initialQuery: searchQuery.trim(),
+					useStandardBack: true,
+				} );
+				transferDomainStepURL += `/${ selectedSite.slug }?${ query }`;
+			}
+		}
 
-		return this.renderOptionContent( {
-			image,
-			title,
-			reasons,
-			onClick: this.goToMapDomainStep,
-			buttonText,
-			isPrimary: false,
-			learnMore,
-		} );
+		page( transferDomainStepURL );
 	};
 
-	render() {
-		const { isSignupStep, translate } = this.props;
-
-		const header = ! isSignupStep && (
-			<HeaderCake onClick={ this.goBack }>
-				{ this.props.translate( 'Use My Own Domain' ) }
-			</HeaderCake>
-		);
-
-		return (
-			<div className="use-your-domain-step">
-				{ header }
-				<QueryProducts />
-				<div className="use-your-domain-step__content">
-					{ this.renderSelectTransfer() }
-					{ this.renderSelectMapping() }
```

Comment:
Refactored these two render functions (and the underlying `renderOptionContent`) to an internal `UseYourDomainStepContent` component